ic10emu/src/grammar.rs

```diff
@@ -368,27 +370,25 @@ impl Operand {
                 target,
             }) => ic.get_register(indirection, target),
             Operand::Number(num) => Ok(num.value()),
-            Operand::LogicType(lt) => lt
-                .get_str("value")
-                .map(|val| val.parse::<u16>().unwrap() as f64)
-                .ok_or(interpreter::ICError::TypeValueNotKnown),
-            Operand::SlotLogicType(slt) => slt
-                .get_str("value")
-                .map(|val| val.parse::<u16>().unwrap() as f64)
-                .ok_or(interpreter::ICError::TypeValueNotKnown),
-            // default to using LogicType when converting to value
-            Operand::LogicOrSlotLogicType(lt, _) => lt
-                .get_str("value")
-                .map(|val| val.parse::<u16>().unwrap() as f64)
-                .ok_or(interpreter::ICError::TypeValueNotKnown),
-            Operand::BatchMode(bm) => bm
-                .get_str("value")
-                .map(|val| val.parse::<u8>().unwrap() as f64)
-                .ok_or(interpreter::ICError::TypeValueNotKnown),
-            Operand::ReagentMode(rm) => rm
-                .get_str("value")
-                .map(|val| val.parse::<u8>().unwrap() as f64)
-                .ok_or(interpreter::ICError::TypeValueNotKnown),
+            Operand::Type {
+                logic_type,
+                slot_logic_type,
+                batch_mode,
+                reagent_mode,
+                identifier: _,
+            } => {
+                if let Some(lt) = logic_type {
+                    Ok(lt.get_str("value").unwrap().parse::<u16>().unwrap() as f64)
+                } else if let Some(slt) = slot_logic_type {
+                    Ok(slt.get_str("value").unwrap().parse::<u16>().unwrap() as f64)
+                } else if let Some(bm) = batch_mode {
+                    Ok(bm.get_str("value").unwrap().parse::<u8>().unwrap() as f64)
+                } else if let Some(rm) = reagent_mode {
+                    Ok(rm.get_str("value").unwrap().parse::<u8>().unwrap() as f64)
+                } else {
+                    Err(interpreter::ICError::TypeValueNotKnown)
+                }
+            }
```

Comment:
should not use unwrap here for the get i think, I'm pretty sure it'll always succeed but to be on the safe side